Sprint 9 solution http api

PaperAlex opened the pull request.

PaperAlex committed:
Техническое задание проекта №6 ver 1

PaperAlex committed:
Техническое задание проекта №6 ver 1

PaperAlex committed:
Техническое задание проекта №6 ver 1.1

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Переименовать пакеты с тестами

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Поставить пустую строку перед классом

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.2
1. Переназвать переменную inMemoryHistoryManager
2. Использовать интефейс List
3. Добавить новый метод getSubTasksIdByEpicId

PaperAlex committed:
Техническое задание проекта №6 ver 1.2
1. Переназвать переменную inMemoryHistoryManager
2. Использовать интефейс List
3. Добавить новый метод getSubTasksIdByEpicId

PaperAlex committed:
Техническое задание проекта №7 ver 0.0

PaperAlex committed:
Техническое задание проекта №7 ver 1.0

PaperAlex committed:
Техническое задание проекта №7 ver 1.0

PaperAlex committed:
Merge branch 'main' into sprint_7-solution-in-file-manager

PaperAlex committed:
Техническое задание проекта №7 ver 1.1 checkStyle

PaperAlex committed:
Merge remote-tracking branch 'origin/sprint_7-solution-in-file-manager' into sprint_7-solution-in-file-manager

PaperAlex committed:
Техническое задание проекта №7 ver 1.2
- завел исключение ManagerLoadException
- сделал FileBackedTaskManager приватным

PaperAlex committed:
Техническое задание проекта №7 ver 1.2
- завел исключение ManagerLoadException
- сделал FileBackedTaskManager приватным

PaperAlex committed:
Техническое задание проекта №8 ver 1

PaperAlex committed:
Merge branch 'main' into sprint_8-solution-time-and-duration

PaperAlex committed:
Техническое задание проекта №8 ver 1

PaperAlex committed:
Merge remote-tracking branch 'origin/sprint_8-solution-time-and-duration' into sprint_8-solution-time-and-duration

PaperAlex committed:
Техническое задание проекта №8 ver 1

PaperAlex committed:
Техническое задание проекта №8 ver 1.2
1. поправил ManagerTimeException
2. поправил Epic

PaperAlex committed:
Техническое задание проекта №8 ver 1.2
1. поправил ManagerTimeException
2. поправил Epic

PaperAlex committed:
Техническое задание проекта №8 ver 1.2
1. поправил ManagerTimeException
2. поправил Epic
3. поправил Task

PaperAlex committed:
Техническое задание проекта №9 ver 1.0

PaperAlex committed:
Техническое задание проекта №9 ver 1.0

PaperAlex committed:
Merge branch 'main' into sprint_9-solution-http-api

PaperAlex committed:
Техническое задание проекта №9 ver 1.0

PaperAlex committed:
Merge remote-tracking branch 'origin/sprint_9-solution-http-api' into sprint_9-solution-http-api

EugeneLenkevich commented on test/managertest/HttpTaskServerTest.java:
```diff
@@ -0,0 +1,405 @@
+
+    TaskManager taskManager = Managers.getDefaultFile();
+    HttpTaskServer taskServer = new HttpTaskServer(taskManager);
+    Gson gson = new GsonBuilder()
```
А давай все таки создание gson вынесем в отдельный метод, например, в отдельном классе чтобы у нас была только одна точка создания и настройки объекта-сериализатора, иначе будет трудновато потом такой код менять, можно что то пропустить :)

EugeneLenkevich commented on test/managertest/HttpTaskServerTest.java:
```diff
@@ -0,0 +1,405 @@
+            .registerTypeAdapter(LocalDateTime.class, new LocalDateTimeAdapter())
+            .create();
+
+    public HttpTaskServerTest() throws IOException {
```
Конструктор в тесте не нужен, кстати

PaperAlex replied:
Легкая паника, когда без конструктора все перестало работать.

EugeneLenkevich replied:
Но сейчас же вроде нормально работает :)

По идее от конструктора не должно тут ничего зависеть, junit по другому немного работает, есть методы before и after и сами тесты, концептуально это как бы не совсем объект, скорее набор описаний кейсов, хотя технически класс, да

EugeneLenkevich commented on test/managertest/HttpTaskServerTest.java:
```diff
@@ -0,0 +1,405 @@
+        HttpClient client = HttpClient.newHttpClient();
+        URI url = URI.create("http://localhost:8080/test");
+
+        HttpRequest request = HttpRequest.newBuilder()
```
```
        HttpRequest request = HttpRequest.newBuilder()
                .uri(url)
                .GET()
                .build();
```

Тест - по сути описание некоего бизнес-сценария, своеобразная документация по тому как наша логика должна работать, а вот технический код несколько затрудняет чтение этой логики, давай попробуем вынести весь этот код создания запросов в отдельные вспомогательные методы и посмотрим как наш код будет после этого выглядеть :)

EugeneLenkevich reviewed (changes requested):
Привет, Алексей! Спасибо за ПР

Это, конечно, муторное дело разбирать вручную запросы, но потом это будет проще, современные фреймворки делают это гораздо проще :)

От очень хочется поработать с тестами, отметил несколько моментов, посмотри комментарии, пожалуйста :)

PaperAlex committed:
Техническое задание проекта №9 ver 1.2
1. Добавил метод getGson
2. Домабил методы запросов в тесты
3. Удалил конструктор HttpTaskServerTest

EugeneLenkevich reviewed (approved):
Да, примерно так, спасибо :)
